src/lib/icon/icon.ts

```diff
@@ -164,21 +163,26 @@ export class MdIcon extends _MdIconMixinBase implements OnChanges, OnInit, CanCo
 
   private _setSvgElement(svg: SVGElement) {
     const layoutElement = this._elementRef.nativeElement;
-    // Remove existing child nodes and add the new SVG element.
-    // We would use renderer.detachView(Array.from(layoutElement.childNodes)) here,
-    // but it fails in IE11: https://github.com/angular/angular/issues/6327
-    layoutElement.innerHTML = '';
+
+    // Remove existing child nodes and add the new SVG element. Note that we can't
+    // use innerHTML, because IE will throw if the element has a data binding.
+    for (let child of layoutElement.childNodes) {
```

Comment:
TpyeScript will compile this to a `for` loop that checks `childNodes.length` on every iteration. I know that used to be something to avoid, but not sure if it's necessary any more.